### ISSUE TITLE
Add support for mod tags in client and infrastructure

### ISSUE THREAD

**Issue by politas** [KSP-CKAN/CKAN#2155]
As per #2034, we now have the ability to add tags to mods to make it easier for users to find similar groups of mods.

We need to:

- [ ] add these tags to the mod info in the GUI
- [ ] add a way to filter by tag in the GUI, preferably including a drop down selector
- [ ] add subcommands / options to filter by tag in the CLI
- [ ] validate tags in CKAN-meta and NetKAN PRs 
- [ ] Add tests for invalid tag entries

Please update this list with other features we need to utilise tags properly

**Comment by dbent** [comment]
Also: decide on list of officially recommended tags and publish it somewhere prominent.

**Comment by politas** [comment]
Would it be worth including a list of official tags in the CKAN-meta repo?

**Comment by dbent** [comment]
> Would it be worth including a list of official tags in the CKAN-meta repo?

I don't think so, I think that "privileges" those tags too much. Hell, even calling them "official" does that as well, probably *recommended* or *suggested* tags is better. In my opinion, the only reason to even have them is to avoid the situation where half of some mods are listed as `graphical` mods and the other half are listed as `visual`. Instead they can all be listed as the *recommended* `eye-candy` tag (not an actual suggestion by the way).

Basically, avoiding a bunch of synonymous or similar tags by canonicalizing a few in preference to others. I think just a list on the wiki is more than adequate for that.

**Comment by HebaruSan** [comment]
Can we start adding tag data to NetKAN? It would be easier to work on UI changes if there was some data available for testing.

**Comment by politas** [comment]
@KSP-CKAN/wranglers , can someone start a Wiki page for suggested tags and perhaps start adding tags to .netkan files?

**Comment by smattiso** [comment]
@politas Something like this?
https://github.com/KSP-CKAN/CKAN/wiki/Suggested-Tags
